svglib/fonts.py

```diff
@@ -0,0 +1,199 @@
+    """
+    def __init__(self):
+        self.log = logging.getLogger('{}.{}'.format(__name__, self.__class__.__name__))
+        """ 
+        the map has the form:
+        internal_name as key, and every entry : {'svg_family': 'family_name', 'svg_weight': 'font-weight', 'svg_style': 'font-style', 'rlgFont': 'rlgFontName'}
+        """
+        self._map = {} # for faster searching we use internal keys for finding the matching font
+
+        self.register_default_fonts()
+
+    @staticmethod
+    def build_internal_name(family, weight='normal', style='normal'):
+        """
+        If the weight or style is given, append the capitalized weight and style to the font name.
+        E.g. family="Arial", weight="bold" and style="italic" then the internal name would be "Arial-BoldItalic", this mimics the default fonts naming schema
+        """
+        result_name = family
+        if weight != 'normal' or style != 'normal':
+            result_name += '-'
+        if weight != 'normal':
+            if type(weight) == int:
+                result_name += f'{weight}'
+            else:
+                result_name += weight.lower().capitalize()
+        if style != 'normal':
+            result_name += style.lower().capitalize()
+        return result_name
+
+    @staticmethod
+    def guess_font_filename(basename, weight='normal', style='normal', extension='ttf'):
+        """
+        Try to guess the actual font filename depending on family, weight and style, 
+        this works at least for windows on the "default" fonts like, Arial, courier, Times New Roman etc.
+        """
+        prefix = ''
+        is_bold = (weight.lower() == 'bold')
+        is_italic = (style.lower() == 'italic')
+        if is_bold and not is_italic:
+            prefix = 'bd'
+        elif is_bold and is_italic:
+            prefix = 'bi'
+        elif not is_bold and is_italic:
+            prefix = 'i'
+        filename = f'{basename}{prefix}.{extension}'
+        return filename
+
+    @staticmethod
+    def use_fontconfig(font_name, weight='normal', style='normal'):
+        NOT_FOUND = (None, False)
+        # Searching with Fontconfig
+        try:
+            pipe = subprocess.Popen(
+                ['fc-match', '-s', '--format=%{file}\\n', font_name],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            output = pipe.communicate()[0].decode(sys.getfilesystemencoding())
+            font_path = output.split('\n')[0]
+            registerFont(TTFont(font_name, font_path))
+        except OSError:
+            return NOT_FOUND 
+        except TTFError:
+            return NOT_FOUND
+        # Fontconfig may return a default font totally unrelated with font_name
+        exact = font_name.lower() in os.path.basename(font_path).lower()
+        return font_name, exact
+
+    def register_default_fonts(self):
```

Comment:
In the current solution the result of register_font can be used to determine if the font was successfull registered. Moving the actual registration in reportlab to a later point would also move the possible errors to the actual conversion process. I personally think that handling those error during the earliest possible time would be better.

Reply:
I simply thought, that currently a user would just register those fonts which are actually needed. If he uses the global font map for all conversions the font would only be registered once anyway. The default fonts dont register any font in reportlab at all, since those are just mappings from svg fonts to one of the fifteen standard reportlab fonts.  But you are right this can be addressed in a separate path.